src/filter.js

```diff
@@ -0,0 +1,241 @@
+// Change State hook of App to change the markers on the map
+
+import { getSetTaps } from "./useTaps"
+
+let buttonValue = {
+    Distance: false,
+    Rating: false,
+    Name: false,
+    Partner: false,
+    Public: false,
+    Spring: false,
+    Cold: false,
+    Hot: false,
+    Filtered: false,
+    Tap: false,
+    Yourself: false,
+    Staff: false,
+    WiFi: false,
+    Toilet: false,
+    Accessible: false,
+    Space: false,
+}
+
+export function createFilter(map, maps) {
+    let clickBool = false;
+    let control = document.createElement('div');
+    const filterButtonUI = menuButtonUi();
+    control.appendChild(filterButtonUI);
+    map.controls[maps.ControlPosition.RIGHT_BOTTOM].push(control);
+
+
+    const filterBoxDiv = document.createElement("div");
+    filterBoxDiv.style.visibility = "hidden";
+
+    const filterUI = filterBoxUi(clickBool, filterBoxDiv);
+    filterParametersUi(filterUI);
+
+    filterBoxDiv.appendChild(filterUI);
+    map.controls[maps.ControlPosition.TOP_LEFT].push(filterBoxDiv);
+
+    filterButtonUI.addEventListener("click", () => {
+        if (!clickBool) {
```

Comment:
Remove the Div element from the map when the menu is closed

Reply:
The element is currently hidden when it is closed. Therefore, we can't add another element or move the map at his position.

Reply:
This issue has been resolved in the following commit: [63ea6c9](https://github.com/mymizu/mymizu-web/pull/21/commits/63ea6c97ce87dcb793a4ad7fab00b1ed2081d4fb)